src/components/overlays/OverlayShell.vue

```diff
@@ -0,0 +1,297 @@
+<!--
+//  OverlayShell.vue
+//
+//  Created by Kalila L. & Heather Anderson on May 13th, 2021.
+//  Copyright 2021 Vircadia contributors.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+-->
+
+<style lang="scss" scoped>
+    .outer {
+        position: absolute;
+    }
+
+    div.title {
+        overflow: hidden;
+        white-space: nowrap;
+        text-overflow: ellipsis;
+    }
+
+    .bar {
+        cursor: default;
+        user-select: none;
+    }
+
+    div.resize {
+        position: absolute;
+        opacity: 0;
+        background-color: red;
+    }
+    div.resize-top, div.resize-nw, div.resize-ne, div.resize-bottom, div.resize-sw, div.resize-se {
+        height: 15px;
+    }
+    div.resize-top, div.resize-nw, div.resize-ne {
+        top: -10px;
+    }
+    div.resize-bottom, div.resize-sw, div.resize-se {
+        bottom: -10px;
+    }
+    div.resize-top, div.resize-bottom {
+        width: 100%;
+        cursor: ns-resize;
+    }
+    div.resize-left, div.resize-nw, div.resize-ne, div.resize-right, div.resize-sw, div.resize-se {
+        width: 15px;
+    }
+    div.resize-left, div.resize-nw, div.resize-sw {
+        left: -10px;
+    }
+    div.resize-right, div.resize-ne, div.resize-se {
+        right: -10px;
+    }
+    div.resize-left, div.resize-right {
+        height: 100%;
+        cursor: ew-resize;
+    }
+    div.resize-nw, div.resize-se {
+        cursor: nwse-resize;
+    }
+    div.resize-ne, div.resize-sw {
+        cursor: nesw-resize;
+    }
+</style>
+
+<template>
+    <q-card
+        class="outer column no-wrap items-stretch"
+        @mouseenter="hovered=true"
+        @mouseleave="hovered=false"
+        @mousedown="$emit('overlay-action', 'select')"
+        :style="{
+            // Dimensions
+            height: height + 'px', // Should these two be a string so that we can define vh or whatever at will?
```

Comment:
```suggestion
            height: height + 'px', // TODO: Should these two be a string so that we can define vh or whatever at will?
```
Or "FIXME".

Reply:
Note that "height" is a number that is changed during any resize actions.  We may end up deciding to change the units involved but it is unlikely to ever be a string.

Reply:
Ah so we'd have to decide what units we were using from the get go, px is likely the easiest to handle though.